DIP/DIP/Controllers/PaymentController.cs

```diff
@@ -0,0 +1,31 @@
+﻿using DIP.Interfaces;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace DIP.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class PaymentController : Controller
+    {
+        private IPayment _paymentService;
+        public PaymentController (IPayment paymentService)
+        {
+            _paymentService = paymentService;
+        }
+        [HttpPost]
+        public ActionResult Pay(int total, double discount)
+        {
+            try
```

Comment:
remove try catch block, and handle the exceptions globally.
you can use the last session as a reference